### PR TITLE
Upgrade

### LINKED ISSUE #64: LDAP integration
The upcoming version 0.5 of Galene will have support for external authorisation servers.  A working but slightly incomplete version of that code is in the branch "auth" of the Galene repository.  It is described here: https://lists.galene.org/galene/87tugzwezl.wl-jch@irif.fr

It would certainly be desirable if that support could be used to better integrate Galene with Yunohost.  Other people who have expressed interest in working on an LDAP authorisation server for Galene are @erdnaxe and @georgesk.  Please see also the discussion at https://github.com/jech/galene/pull/106.

### LINKED ISSUE #79: Passwords with quotes breaks the JSON
### Describe the bug

When installing the application, if the password contains quotes, it breaks the configuration json file (`/home/yunohost.app/galene/groups/GROUP.json`)

- Install Galene using a password containing quotes (like `1"2"3"4"5`)
- After the installation, you should not be able to join the group filled during the installation
- Also you should get a broken JSON in `/home/yunohost.app/galene/groups/GROUP.json` due to the quotes not being escaped

### Logs

```
mars 13 21:31:51 domain.tld systemd[1]: Started Galène: videoconferencing server.
mars 13 21:31:51 domain.tld galene[23076]: 2022/03/13 21:31:51 Starting built-in TURN server on XX.XX.XX.XX:1195
mars 13 21:31:51 domain.tld galene[23076]: 2022/03/13 21:31:51 data/config.json: invalid character '<character here>' after object key:value pair
mars 13 21:31:51 domain.tld galene[23076]: 2022/03/13 21:31:51 Group file /home/yunohost.app/galene/groups/skeptikon.json: invalid character '<character here>' after object key:value pair
mars 13 21:31:51 domain.tld galene[23076]: 2022/03/13 21:31:51 Relay test successful in 30.98294ms, RTT = 264.613µs
mars 13 21:37:22 domain.tld galene[23076]: 2022/03/13 21:37:22 group.Add: invalid character '<character here>' after object key:value pair
```

Thanks for your work! :pray:

### PR DESCRIPTION
**Pull request by yalh76**
## Problem
- *Fix #64*
- *Fix #79*

## Solution

- *Apply last example_ynh*
- *Implement autoupdater and switch to building with Go*
- *Implement LDAP authentication*
- *Tweak the group page*

## PR Status

- [ ] Code finished and ready to be reviewed/tested
- [ ] The fix/enhancement were manually tested (if applicable)

## Automatic tests

Automatic tests can be triggered on https://ci-apps-dev.yunohost.org/ *after creating the PR*, by commenting "!testme", "!gogogadgetoci" or "By the power of systemd, I invoke The Great App CI to test this Pull Request!". (N.B. : for this to work you need to be a member of the Yunohost-Apps organization)

### CODE REVIEW

**Comment by yalh76** [comment]
!testme

**Comment by yalh76** [comment]
!testme

**Comment by yalh76** [comment]
!testme

File: conf/groupname-ldap.json [hunk 1]
```diff
@@ -0,0 +1,8 @@
+{
+ 	"description": "Group of YunoHost users",
```
Comment (ericgaspar) [inline conf/groupname-ldap.json:2]:
```suggestion
 	"description": "YunoHost User Group",
```

File: scripts/install [hunk 2]
```diff
@@ -117,6 +124,28 @@ ynh_script_progression --message="Configuring NGINX web server..." --weight=2
...
+# SPECIFIC SETUP
+#=================================================
+# BUILD APP
+#=================================================
+ynh_script_progression --message="Building app..." --weight=1
+
+ynh_exec_warn_less ynh_install_go --go_version=$go_version
+ynh_use_go
+pushd $final_path/build/
+	ynh_exec_as $app CGO_ENABLED=0 $ynh_go build -ldflags='-s -w' -o $final_path/live/
```
Comment (ericgaspar) [inline scripts/install:137]:
```suggestion
	ynh_exec_warn_less ynh_exec_as $app CGO_ENABLED=0 $ynh_go build -ldflags='-s -w' -o $final_path/live/
```

File: scripts/install [hunk 3]
```diff
@@ -117,6 +124,28 @@ ynh_script_progression --message="Configuring NGINX web server..." --weight=2
...
+popd
+pushd $final_path/build_ldap/
+	ynh_exec_as $app CGO_ENABLED=0 $ynh_go build -ldflags='-s -w' -o $final_path/live_ldap/
```
Comment (ericgaspar) [inline scripts/install:140]:
```suggestion
	ynh_exec_warn_less ynh_exec_as $app CGO_ENABLED=0 $ynh_go build -ldflags='-s -w' -o $final_path/live_ldap/
```

Comment (ericgaspar) [reply scripts/install:140]: let's try to get rid of all the warnings

**Review by ericgaspar — CHANGES_REQUESTED** [review]

**Comment by yalh76** [comment]
!testme

**Comment by yalh76** [comment]
oupsi, I forgot to manage upgrade from current version

**Comment by yalh76** [comment]
!testme

**Review by ericgaspar — APPROVED** [review]
Thanks for improving the package!

**Pull request merged**